reconcile/utils/saasherder.py

```diff
@@ -99,75 +105,73 @@ def _validate_saas_files(self):
-                        environment_name
-                    )
-                    # promotion publish channels
-                    promotion = target.get('promotion')
-                    if promotion:
-                        publish = promotion.get('publish')
-                        if publish:
-                            saas_file_promotion_publish_channels.extend(
-                                publish)
-                    # validate target parameters
-                    target_parameters = target['parameters']
-                    if not target_parameters:
+            resource_template_name = resource_template['name']
+            target_namespace = target['namespace']
+            namespace_name = target_namespace['name']
+            cluster_name = target_namespace['cluster']['name']
+            environment = target_namespace['environment']
+            environment_name = environment['name']
+            # unique saas file and env name combination
+            self._check_saas_file_env_combo_unique(
+                saas_file_name,
+                environment_name
+            )
+            # promotion publish channels
+            promotion = target.get('promotion')
+            if promotion:
+                publish = promotion.get('publish')
+                if publish:
+                    saas_file_promotion_publish_channels.extend(
+                        publish)
+            # validate target parameters
+            target_parameters = target['parameters']
+            if not target_parameters:
+                continue
+            target_parameters = json.loads(target_parameters)
+            environment_parameters = environment['parameters']
+            if not environment_parameters:
+                continue
+            environment_parameters = \
+                json.loads(environment_parameters)
+            msg = \
+                f'[{saas_file_name}/{resource_template_name}] ' + \
+                'parameter found in target ' + \
+                f'{cluster_name}/{namespace_name} ' + \
+                f'should be reused from env {environment_name}'
+            for t_key, t_value in target_parameters.items():
+                if not isinstance(t_value, str):
+                    continue
+                # Check for recursivity. Ex: PARAM: "foo.${PARAM}"
+                replace_pattern = '${' + t_key + '}'
+                if replace_pattern in t_value:
+                    logging.error(
+                        f'[{saas_file_name}/{resource_template_name}] '
+                        f'recursivity in parameter name and value '
+                        f'found: {t_key}: "{t_value}" - this will '
+                        f'likely not work as expected. Please consider'
+                        f' changing the parameter name')
+                    self.valid = False
+                for e_key, e_value in environment_parameters.items():
+                    if not isinstance(e_value, str):
```

Comment:
Stylistic disagreement here, I prefer to have all conditions grouped in a single `if` statement, these `continue`s break my reading flow, it feels like I'm biking over a very bumpy road.

Reply:
Just a comment, not a blocker.